src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -11007,29 +11007,16 @@ private ExpressionSyntax ParseExpressionContinued(ExpressionSyntax leftOperand,
             return leftOperand;
         }
 
-        private DeclarationExpressionSyntax ParseDeclarationExpression(ParseTypeMode mode, MessageID feature, bool isScoped)
+        private DeclarationExpressionSyntax ParseDeclarationExpression(ParseTypeMode mode, bool isScoped)
         {
-            SyntaxToken scopedKeyword = null;
-
-            if (isScoped)
-            {
-                scopedKeyword = EatContextualToken(SyntaxKind.ScopedKeyword);
-            }
-
-            TypeSyntax type = this.ParseType(mode);
-
-            if (scopedKeyword != null)
-            {
-                type = _syntaxFactory.ScopedType(scopedKeyword, type);
-            }
-
-            var designation = ParseDesignation(forPattern: false);
-            if (feature != MessageID.None)
-            {
-                designation = CheckFeatureAvailability(designation, feature);
-            }
```

Comment:
here's what causes us to only report on the outer expr.  The inner designations (recursively called from ParseDesignation don't get a feature message on them).